joblib/test/test_memory.py

```diff
@@ -34,50 +33,15 @@ def f(x, y=1):
 
 
 ###############################################################################
-# Test fixtures
-env = dict()
-
-
-def setup_module():
-    """ Test setup.
-    """
-    cachedir = mkdtemp()
-    env['dir'] = cachedir
-    if os.path.exists(cachedir):
-        shutil.rmtree(cachedir)
-    # Don't make the cachedir, Memory should be able to do that on the fly
-    print(80 * '_')
-    print('test_memory setup (%s)' % env['dir'])
-    print(80 * '_')
-
-
-def _rmtree_onerror(func, path, excinfo):
-    print('!' * 79)
-    print('os function failed: %r' % func)
-    print('file to be removed: %s' % path)
-    print('exception was: %r' % excinfo[1])
-    print('!' * 79)
-
-
-def teardown_module():
-    """ Test teardown.
-    """
-    shutil.rmtree(env['dir'], False, _rmtree_onerror)
-    print(80 * '_')
-    print('test_memory teardown (%s)' % env['dir'])
-    print(80 * '_')
-
-
-###############################################################################
 # Helper function for the tests
-def check_identity_lazy(func, accumulator):
+def check_identity_lazy(func, accumulator, tmpdir):
```

Comment:
Nitpick: I would replace `tmpdir` by `cachedir`, this feels more clear. You will need to call `check_identity_lazy(func, accumulator, tmpdir.strpath)` in a few places.